### PR TITLE
Add --no-color option to disable color output

### LINKED ISSUE #890: zarf command line execution error logging and log file readability
**Is your feature request related to a problem? Please describe.**
When executing _zarf package deploy_ from a command line, the erroring reporting seems inconsistent. 
Example 1:
A deployment.yaml is missing spec.selector. 
zarf output:
Screen Shot 2022-10-12 at 8 43 51 AM

Example 2:
A deployment.yaml is missing spec.template.metadata
zarf output: 
Screen Shot 2022-10-12 at 8 47 23 AM
 
In both scenarios, when trying to review/troubleshoot using the associated log file with linux tools like vi, nano or simply cat the log file piped to less (cat /var/folders/bh/l8qqtz8x57j5vjd8sd33sxt40000gn/T/zarf-2022-10-12-08-46-18-3119412142.log | less) the log file is difficult to read due to the color flags. 
Screen Shot 2022-10-12 at 9 09 17 AM

**Describe the solution you'd like**
For example 1, similar logging output to example 2.
The ability to use basic linux tools to easily review and examine the log file for troubleshooting.

**Describe alternatives you've considered**
Manually testing yaml files before building a zarf package (i.e. kubectl apply -f deployment.yaml) to avoid the "WARNING Unable to complete helm chart install/update" messages and application deployment failure.
Don't currently have a good alternative to viewing the logs. I can investigate other editing tools.

**Additional context**

### PR DESCRIPTION
**Pull request by caesarshift**
## Description

Add --no-color option to disable color output

## Related Issue

Fixes #890

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

**Comment by caesarshift** [comment]
I created this as a draft MR in order to get feed back on [this note](https://github.com/defenseunicorns/zarf/issues/890#issuecomment-1625619936)

**Comment by YrrepNoj** [comment]
Replying to this comment you left on the issue: https://github.com/defenseunicorns/zarf/issues/890#issuecomment-1625619936


The colored log message that says where we're loading the config file from is happening in the [`initViper(...)`](https://github.com/defenseunicorns/zarf/blob/ad5f462135838d11d9ce1080bf4f3b20b35315ad/src/cmd/viper.go#L83C3-L83C3) function.

That `initViper(...)` is called by the [init function](https://github.com/defenseunicorns/zarf/blob/ad5f462135838d11d9ce1080bf4f3b20b35315ad/src/cmd/root.go#L79) for `root.go`. Since it's called by an init function, we can not rely on any configuration to be set yet.

If we want to make it so the log message for which config file is being used follows the same configuration as the rest of the logs, we could just move [this line](https://github.com/defenseunicorns/zarf/blob/ad5f462135838d11d9ce1080bf4f3b20b35315ad/src/cmd/viper.go#L124C13-L124C13) to after we have called `cliSetup(...)`

**Comment by caesarshift** [comment]
> Replying to this comment @caesarshift left on the issue: [#890 (comment)](https://github.com/defenseunicorns/zarf/issues/890#issuecomment-1625619936)
> 
> The colored log message that says where we're loading the config file from is happening in the [`initViper(...)`](https://github.com/defenseunicorns/zarf/blob/ad5f462135838d11d9ce1080bf4f3b20b35315ad/src/cmd/viper.go#L83C3-L83C3) function.
> 
> That `initViper(...)` is called by the [init function](https://github.com/defenseunicorns/zarf/blob/ad5f462135838d11d9ce1080bf4f3b20b35315ad/src/cmd/root.go#L79) for `root.go`. Since it's called by an init function, we can not rely on any configuration to be set yet.
> 
> If we want to make it so the log message for which config file is being used follows the same configuration as the rest of the logs, we could just move [this line](https://github.com/defenseunicorns/zarf/blob/ad5f462135838d11d9ce1080bf4f3b20b35315ad/src/cmd/viper.go#L124C13-L124C13) to after we have called `cliSetup(...)`

Based on your comment, I've added a follow-on [commit](https://github.com/defenseunicorns/zarf/pull/1889/commits/606258566b3034ce0e56e27f377e183af21c8426) that moves the print statements into the cli setup (moving it into the function preserves the order of the output to precede the current log level output)

File: docs/2-the-zarf-cli/100-cli-commands/zarf_tools.md
Comment (Noxsios) [inline docs/2-the-zarf-cli/100-cli-commands/zarf_tools.md:1]: the `--no-color` flag should probably not be passed to `zarf tools` as these sub tools would control their own color (example - `k9s`), so this flag would not do anything + might cause confusion.

Comment (caesarshift) [reply docs/2-the-zarf-cli/100-cli-commands/zarf_tools.md:1]: I agree. However, this would also apply to all the other global flags too, right?

Comment (Noxsios) [reply docs/2-the-zarf-cli/100-cli-commands/zarf_tools.md:1]: Correct, I believe `addHiddenDummyFlag(toolCmd, "no-color")` in `src/cmd/internal.go` is what I am thinking of

Comment (Racer159) [reply docs/2-the-zarf-cli/100-cli-commands/zarf_tools.md:1]: For the other commands this is controlled via the `isVendorCmd` function - some tools (the one's Zarf itself has) would get the `--no-color` option but the option should not show on vendored commands like `kubectl` or `syft`.

(Notice how this section exists now on the vendored commands:)

```
## Options inherited from parent commands

---
      --no-color   Disable colors in output
---
```

Comment (Racer159) [reply docs/2-the-zarf-cli/100-cli-commands/zarf_tools.md:1]: @caesarshift just following up on the above to see if there was anything we could help with

Comment (caesarshift) [reply docs/2-the-zarf-cli/100-cli-commands/zarf_tools.md:1]: Ah, @Noxsios - thanks for clarifying. I  thought you were talking about the cli help documentation that is output when you run the tool, and I couldn't see how I was treating the flag any differently than the other global flags. But this is about the generated docs. Apologies for my confusion. I've made the requested change and regenerated the documentation.

**Comment by caesarshift** [comment]
FYSA, due to a merge conflict in src/config/config.go, I also rebased the branch.

**Review by Racer159 — APPROVED** [review]
lgtm

**Review by cmwylie19 — APPROVED** [review]
lgtm

**Review by Racer159 — CHANGES_REQUESTED** [review]

File: src/cmd/viper.go [hunk 1]
```diff
@@ -116,6 +117,11 @@ func initViper() {
...
 	v.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
 	v.AutomaticEnv()
+}
+
+func printViperConfigUsed() {
+	// Confirm initialization
+	initViper()
 
 	// Optional, so ignore errors
 	err := v.ReadInConfig()
```
[the remark this answers is not in the document]
Comment (Racer159) [reply src/cmd/viper.go]: Yeah that is likely a fine place for now - there is a big viper refactor in #1898 that will likely make that make a bit more sense (decoupling viper from the root command) but this should be fine for now.

**Review by Racer159 — APPROVED** [review]
lgtm

**Review by cmwylie19 — APPROVED** [review]
lgtm

**Review by cmwylie19 — APPROVED** [review]
lgtm

**Pull request merged**